src/node.cc

```diff
@@ -877,8 +877,7 @@ Local<Value> ErrnoException(Isolate* isolate,
 
   Local<String> path_string;
   if (path != nullptr) {
-    // FIXME(bnoordhuis) It's questionable to interpret the file path as UTF-8.
-    path_string = String::NewFromUtf8(env->isolate(), path);
+    path_string = FIXED_ONE_BYTE_STRING(env->isolate(), path);
```

Comment:
FWIW, there are edge cases where even this is going to be inadequate. Those are the same edge cases that led me to add the ability to pass `Buffer` instances to `fs` APIs... specifically, on posix systems, it's possible for a path string to contain *multiple* encodings... for instance, parent directory in ASCII, directory in Windows-1250, UTF-8 filename, etc. Unfortunately it's not theoretical either... actually had this happen.

May be worthwhile deprecating `path_string` and moving to a `Buffer` instance.